### PR TITLE
Set up opam workflow

### PR DESCRIPTION
**Pull request by tekknolagi**
- [x] Makefile & topkg
- [x] Docs
- [x] CI
- [ ] Update `docs/` folder

### CODE REVIEW

File: opam [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+bug-reports: "https://github.com/reasonml/ReasonProject/issues"
+dev-repo: "git://github.com/reasonml/ReasonProject.git"
+tags: [ "reason" "example" ]
+substs: [ "pkg/META" ]
+build: [
+  [make "build"]
+]
+depends: [
+  "topkg"         {>=  "0.8.1" & < "0.9"}
+  "reason-parser" {= "1.13.3"}
```
Comment (yunxing) [inline opam]: This should depend on Reason?

Comment (tekknolagi) [inline opam]: Yep, sorry

**Comment by yunxing** [comment]
Please update the CI to use opam, thanks!

**Comment by tekknolagi** [comment]
Why does this not install opam 1.2 like the Reason CircleCI build? Who knows????

**Comment by tekknolagi** [comment]
@yunxing CircleCI passes. Travis fails because bad config. Let's see if I fixed it...

**Pull request merged**